### PR TITLE
Code and test for detecting cyclic graph added #12

### PR DESCRIPTION
**Pull request by yatingupta10**

### CODE REVIEW

File: pydsa/tests/test_detecting_cyclicgraph.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from pydsa.detecting_cyclicgraph import detecting_cyclicgraph
+
+def test_detecting_cyclicgraph():
+	assert detecting_cyclicgraph({1: (2,), 2: (3,), 3: (1,)})==True
+	assert detecting_cyclicgraph({1: (2,), 2: (3,), 3: (4,)})==False
```
Comment (CuriousLearner) [inline pydsa/tests/test_detecting_cyclicgraph.py]: Please leave a line at the end of file

File: pydsa/detecting_cyclicgraph.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+            return False
+        visited.add(vertex)
+        path.add(vertex)
+        for neighbour in g.get(vertex, ()):
+            if neighbour in path or visit(neighbour):
+                return True
+        path.remove(vertex)
+        return False
+
+    return any(visit(v) for v in g)
```
Comment (CuriousLearner) [inline pydsa/detecting_cyclicgraph.py]: Please leave a line at the end of file

File: pydsa/tests/test_detecting_cyclicgraph.py (hunk 1 above)
Comment (CuriousLearner) [inline pydsa/tests/test_detecting_cyclicgraph.py]: Spaces should be there before operator

**Comment by CuriousLearner** [comment]
Apart from the diff, please check and format for PEP8 in general @yatingupta10

**Comment by yatingupta10** [comment]
done as PEP8 format @CuriousLearner

**Comment by CuriousLearner** [comment]
Refers #122 

cc @aktech

**Pull request closed without merging**